.github/workflows/docs.yaml

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright contributors to the Galasa project
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+name: Galasa documentation build
+
+on:
+  workflow_call:
+
+env:
+  BRANCH: ${{ github.ref_name }}
+
+jobs:
+  build-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the docs source
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            docs
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - run: echo "cache_id=$(date --utc '+%V')" >> $GITHUB_ENV 
+
+      - uses: actions/cache@v4
+        with:
+          key: mkdocs-material-
+          path: .cache
+          restore-keys: |
+            mkdocs-material-
+
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          gradle-version: 8.9
+          gradle-home-cache-excludes: |
+            caches/modules-2/files-2.1/dev.galasa/**
+
+
+      - name: Install python packages
+        run: |
+          pip install mkdocs-material 
+
+      - name: Build and package the docs
+        working-directory: ./docs
+        run : |
+          set -o pipefail
+          gradle -b build.gradle publish --info --warning-mode all \
+          --no-daemon --console plain \
+          -PsourceMaven=/home/runner/.m2/repository \
+          -PcentralMaven=https://repo.maven.apache.org/maven2/ \
+          -PtargetMaven=${{ github.workspace }}/docs/repo 2>&1 | tee build-docs.log      
+
+      - name: Upload build log
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-gradle-build-log
+          path: docs/build-docs.log
+          retention-days: 7
+      
+      - name: Upload docs artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-maven
+          path: docs/repo
+
+      - name: Upload docs site
+        uses: actions/upload-artifact@v4
+        with:
+          name: galasa-docs-site
+          path: docs/build/site
+
+  deploy-docs-pr-preview-to-github-pages:
+
+    if: ${{ (github.ref_name == 'main') }}
+    permissions:
+      contents: write
+
```

Comment:
I think this section can be removed, I don't think it will be needed

Reply:
removed.